### PR TITLE
Update our packaging stuff from the template

### PR DESCRIPTION
**Pull request by Cadair**
This enforces our 3.6 minimum Python version and tidies things up a little

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -31,18 +31,44 @@ auto_use = True
...
-license = BSD
-url = http://sunpy.org/
+license = BSD 2-Clause
+url = https://sunpy.org
 edit_on_github = True
 github_project = sunpy/sunpy
-
-[flake8]
-max-line-length = 100
+install_requires = numpy>=1.11, astropy>=2.0.3>=3.0.2, scipy, pandas>=0.12.0, matplotlib>=1.3
```
Comment (nabobalis) [inline setup.cfg]: One day be nice to either parse the reqs.txt or define it all here.

Comment (nabobalis) [reply setup.cfg]: Do we want them anymore?

Comment (Cadair) [reply setup.cfg]: I thought RTD uses them at least?

Comment (nabobalis) [reply setup.cfg]: yes it does

**Comment by nabobalis** [comment]
Sugar looks off on travis build.

File: setup.cfg (hunk 1 above)
Comment (nabobalis) [reply setup.cfg]: But maybe we should jsut dump it all into the yaml for RTD and delete those.

File: setup.py [hunk 2]
```diff
@@ -3,57 +3,64 @@
...
+PACKAGENAME = metadata.get('package_name', 'sunpy')
 DESCRIPTION = metadata.get('description', 'SunPy: Python for Solar Physics')
 AUTHOR = metadata.get('author', 'The SunPy Community')
-AUTHOR_EMAIL = metadata.get('author_email', 'sunpy@googlegroups.com')
-LICENSE = metadata.get('license', 'BSD 2-Clause')
-URL = metadata.get('url', 'http://sunpy.org')
+AUTHOR_EMAIL = metadata.get('author_email', '')
+LICENSE = metadata.get('license', 'unknown')
+URL = metadata.get('url', 'https://sunpy.org')
+LONG_DESCRIPTION = metadata.get('long_description', DESCRIPTION)
```
[the remark this answers is not in the document]
Comment (vn-ki) [reply setup.py]: Can we somehow make this be our readme? Would look nice on pypi.

File: sunpy/__init__.py [hunk 3]
```diff
@@ -9,26 +9,43 @@
 Homepage: http://sunpy.org
 Documentation: http://docs.sunpy.org/en/stable/
 """
-from __future__ import absolute_import
+# Enforce Python version check during package import.
+# This is the same check as the one at the top of setup.py
+import os
+import sys
+
+__minimum_python_version__ = "3.6"
```
Comment (vn-ki) [inline sunpy/__init__.py]: Can't we read this from setup.cfg or something? If we change minimum to something else, we'll have to change in two places.

**Comment by nabobalis** [comment]
Doc build failed due to numpy version. Fixed in master, so this should be good to go.

File: setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nabobalis) [reply setup.py]: We could using our current readme: https://packaging.python.org/guides/making-a-pypi-friendly-readme/

**Pull request merged**